bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/smarthome/HandlerPercentageController.java

```diff
@@ -0,0 +1,143 @@
+            AmazonEchoControlBindingConstants.BINDING_ID, "percentage");
+
+    // Channel definitions
+    private static final ChannelInfo PERCENTAGE = new ChannelInfo("percentage" /* propertyName */ ,
+            "percentage" /* ChannelId */, CHANNEL_TYPE_PERCENTAGE /* Channel Type */ ,
+            ITEM_TYPE_DIMMER /* Item Type */);
+
+    private @Nullable Integer lastPercentage;
+
+    @Override
+    public String[] getSupportedInterface() {
+        return new String[] { INTERFACE };
+    }
+
+    @Override
+    protected ChannelInfo @Nullable [] findChannelInfos(SmartHomeCapability capability, String property) {
+        if (PERCENTAGE.propertyName.equals(property)) {
+            return new ChannelInfo[] { PERCENTAGE };
+        }
+        return null;
+    }
+
+    @Override
+    public void updateChannels(String interfaceName, List<JsonObject> stateList, UpdateChannelResult result) {
+        Integer percentageValue = null;
+        for (JsonObject state : stateList) {
+            if (PERCENTAGE.propertyName.equals(state.get("name").getAsString())) {
+                int value = state.get("value").getAsInt();
+                // For groups take the maximum
+                if (percentageValue == null) {
+                    percentageValue = value;
+                } else if (value > percentageValue) {
+                    percentageValue = value;
+                }
+            }
+        }
+        if (percentageValue != null) {
+            lastPercentage = percentageValue;
+        }
+        updateState(PERCENTAGE.channelId, percentageValue == null ? UnDefType.UNDEF : new PercentType(percentageValue));
+    }
+
+    @Override
+    public boolean handleCommand(Connection connection, SmartHomeDevice shd, String entityId,
+            SmartHomeCapability[] capabilties, String channelId, Command command) throws IOException {
+        if (channelId.equals(PERCENTAGE.channelId)) {
+            if (containsCapabilityProperty(capabilties, PERCENTAGE.propertyName)) {
+                if (command.equals(IncreaseDecreaseType.INCREASE)) {
+                    Integer lastPercentage = this.lastPercentage;
+                    if (lastPercentage != null) {
+                        int newValue = lastPercentage++;
+                        if (newValue > 100) {
+                            newValue = 100;
+                        }
+                        this.lastPercentage = newValue;
+                        connection.smartHomeCommand(entityId, "setPercentage", PERCENTAGE.propertyName, newValue);
+                        return true;
+                    }
+                } else if (command.equals(IncreaseDecreaseType.DECREASE)) {
+                    Integer lastPercentage = this.lastPercentage;
+                    if (lastPercentage != null) {
+                        int newValue = lastPercentage--;
+                        if (newValue < 0) {
+                            newValue = 0;
+                        }
```

Comment:
Perhaps you could increase/decrease by a more than just a single percent. A change of a single percent is probably unnoticeable to the user so I think 10% is probably a better value.

Reply:
Since according to the forum thread quite a bunch of users already use this enhancements (since month), I wouldprefer to keep it like this.